### ISSUE TITLE
no use of assert

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/normal-modes#24]
Related to #2 (code review):

> - [ ] Assertions should be used appropriately and consistently. Type checking should not just be done in code comments. Use Array.isArray to type check an array.

This sim has zero uses of `assert`.  Combined with #23 (sim fails with assertions enabled), I'm guessing there is lack of understanding about when and how to use assertions.  

Assertions are an important tool that needs to be in every good programmer's toolbox. "Appropriately" generally mean to verify entry/exit conditions for methods, and verify programming assumptions and logic.

**Comment by pixelzoom** [comment]
Since this sim is near completion, it may not be worth the trouble to go back and add assertions. Ask @ariel-phet before proceeding.  

Let me know if you'd like some examples of where assertions would have been appropriate. They are typically valuable for catching programming errors, including:

* checking that constructor and method arguments are of the proper type and satisfy any constraints
* checking that complicated computations result in an expected value
* checking that a method's return value satisfies constraints

Assertions should _not_ be used for checking values that come from outside the program - for example, user inputs or query parameters values.  Things that are not under the control of the programmer typically throw and Error, which is caught and handled further up the chain.

**Comment by Hyodar** [comment]
Alright. 
Is it a good thing if I, at least, add type check assertions for each method and constructor?
I just wrote a simple script to parse each method's jsdoc `@param` fields and add type checks, so this wouldn't be a big trouble.

**Comment by pixelzoom** [comment]
I don't think you need to do that, just keep it in mind for next time.  In other sims (e.g. gas-properties) I've experimented with type check assertions for every parameter, but it seems like overkill.  I think its acceptable to add type checking for parameters where someone is likely to make a mistake, and (more importantly) for argument values that need to satisfy constraints.
